src/ray/common/asio/periodical_runner.cc

```diff
@@ -52,4 +59,30 @@ void PeriodicalRunner::DoRunFnPeriodically(std::function<void()> fn,
   });
 }
 
+void PeriodicalRunner::DoRunFnPeriodically(
+    std::function<void()> fn, boost::posix_time::milliseconds period,
+    boost::asio::deadline_timer &timer,
+    std::shared_ptr<GuardedHandlerStats> handler_stats,
+    std::shared_ptr<GuardedGlobalStats> global_stats,
+    absl::optional<int64_t> enqueue_time) {
+  io_service_.Instrument(fn, handler_stats, global_stats, enqueue_time);
+  timer.expires_from_now(period);
+  enqueue_time = io_service_.Enqueue(handler_stats);
+  timer.async_wait([this, fn, period, &timer, handler_stats, global_stats,
+                    enqueue_time](const boost::system::error_code &error) {
+    if (error == boost::asio::error::operation_aborted) {
+      // `operation_aborted` is set when `timer` is canceled or destroyed.
+      // The Monitor lifetime may be short than the object who use it. (e.g. gcs_server)
+      return;
```

Comment:
Don't we need to call Instrument even on cancel to decrement the cur number of active calls?

Reply:
Great point, I missed that! I'll try exposing a `RecordEnd()` function for this case.